internetarchive/utils.py

```diff
@@ -134,11 +134,18 @@ def new_hook(type, value, traceback):
 
 
 class IterableToFileAdapter:
-    def __init__(self, iterable, size: int):
+    def __init__(self, iterable, size: int, pre_encode: bool = False):
         self.iterator = iter(iterable)
         self.length = size
+        # pre_encode is needed because http doesn't know that it
+        # needs to encode a TextIO object when it's wrapped
+        # in the Iterator from tqdm.
+        # So, this FileAdapter provides pre-encoded output
+        self.pre_encode = pre_encode
 
     def read(self, size: int = -1):  # TBD: add buffer for `len(data) > size` case
+        if self.pre_encode:
+            return next(self.iterator, '').encode("iso-8859-1")
```

Comment:
Do we really want to go with an ancient, obsolete encoding here rather than UTF-8?

Reply:
This was chosen to match the encoding performed by the HTTP library that this PR emulates. I can test it out with UTF-8, but I have more trust that matching the library exactly will best handle all possible cases.

Reply:
There are a few references in the http lib suggesting that ISO 8859-1 is required in parts of the http standard, see:

https://github.com/python/cpython/blob/e87ada48a9e5d9d03f9759138869216df0d7383a/Lib/http/client.py#L246
https://github.com/python/cpython/blob/e87ada48a9e5d9d03f9759138869216df0d7383a/Lib/http/client.py#L1340

so I think it is best to keep it this way, I will add a comment noting the reason though.